### ISSUE TITLE
Bug with JUNKTEST with Random values on MSYS2 and Cygwin

### ISSUE THREAD

**Issue by kentkost** [mouse07410/asn1c#135]
This is an issue that is as old as time when it comes to random values and stdlib.

In the converter-example.c I can't use JUNKTEST on windows.

First because the method on windows are called
**rand** and **srand** and not **random** and **srandom**

An easy fix for this to just add further macros to converter-example.c

```C
#if defined(__WIN32__) && defined(JUNKTEST)
#define random rand
#define srandom srand
#endif
```

The second issue is that in stdlib.h on windows the **RAND_MAX** is a 16 bit number and is defined as **0x7fff**
But for example on [openbsd](https://github.com/openbsd/src/blob/master/include/stdlib.h) it is defined to be as large as **0x7fffffff**

Which means in the file [asn_random_fill.c line 47](https://github.com/mouse07410/asn1c/blob/b93d80fd4ff987a9b6749363b22e6bce7715075d/skeletons/asn_random_fill.c#L47) it will always fail on Windows.

I have made a fix for this that looks like this:

```C
intmax_t
asn_random_between(intmax_t lb, intmax_t rb) {
        ...
        int max= 0xffffff;

#ifdef __WIN32__
        max= RAND_MAX-1;
#endif

        assert(RAND_MAX > max);    /* Seen 7ffffffd! */
        assert(range < intmax_max);

        for(; got_entropy < range;) {
            got_entropy = (got_entropy << 24) | max;
#ifdef HAVE_RANDOM
            value = (value << 24) | (random() % max);
#else
            value = (value << 24) | (rand() % max);
#endif
        }
        ...
```
Basically it if runs Cygwin or MSYS2 then the max value it can generate is 16bit.
I explored options online about redefining RAND_MAX but the general consensus was that it was a bad idea.

Should I make a pull request on this?

**Comment by mouse07410** [comment]
Thank you, and yes please.
